src/bootstrap/doc.rs

```diff
@@ -491,37 +471,115 @@ impl Step for Std {
             })
             .collect::<Vec<_>>();
 
-        // Only build the following crates. While we could just iterate over the
-        // folder structure, that would also build internal crates that we do
-        // not want to show in documentation. These crates will later be visited
-        // by the rustc step, so internal documentation will show them.
-        //
-        // Note that the order here is important! The crates need to be
-        // processed starting from the leaves, otherwise rustdoc will not
-        // create correct links between crates because rustdoc depends on the
-        // existence of the output directories to know if it should be a local
-        // or remote link.
-        let krates = ["core", "alloc", "std", "proc_macro", "test"];
-        for krate in &krates {
-            run_cargo_rustdoc_for(krate);
-            if paths.iter().any(|p| p == krate) {
-                // No need to document more of the libraries if we have the one we want.
-                break;
-            }
-        }
-        builder.cp_r(&out_dir, &out);
+        doc_std(builder, stage, target, &out, &extra_args, &requested_crates);
 
         // Look for library/std, library/core etc in the `x.py doc` arguments and
         // open the corresponding rendered docs.
-        for requested_crate in paths {
-            if krates.iter().any(|k| *k == requested_crate.as_str()) {
+        for requested_crate in requested_crates {
+            if STD_PUBLIC_CRATES.iter().any(|k| *k == requested_crate.as_str()) {
                 let index = out.join(requested_crate).join("index.html");
                 open(builder, &index);
             }
         }
     }
 }
 
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq)]
+pub struct JsonStd {
+    pub stage: u32,
+    pub target: TargetSelection,
+}
+
+impl Step for JsonStd {
+    type Output = ();
+    const DEFAULT: bool = false;
+
+    fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
+        run.alias("rust-json-docs")
```

Comment:
hmm, this is named after the dist component, but that's fairly unusual for anything other than `dist`. I think a nicer CLI interface would be to add a `--json` *flag* to the `doc` subcommand (not a `config.toml` option), which allows us to extend this to the compiler docs in the future if we want to.
```suggestion
                run.all_krates("test").path("library").default_condition(builder.config.docs && builder.config.json_docs)
```

Reply:
If I understood correctly, it should have been done in [c0ab8ba](https://github.com/rust-lang/rust/pull/101799/commits/c0ab8baa8f927beac47ea07a99ec58877e1c14dd)

Reply:
That looks perfect, thanks!